aiven/resource_database.go

```diff
@@ -55,13 +55,20 @@ var aivenDatabaseSchema = map[string]*schema.Schema{
 		ForceNew:         true,
 		DiffSuppressFunc: handleLcDefaults,
 	},
+	"termination_protection": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Prevent database from being deleted. It is recommended to have this enabled for all services.",
```

Comment:
Would be good to mention this is client-side only functionality so the database can be deleted from web console or by direct API call. Also `It is recommended to have this enabled for all services.` is a bit vague recommendation. If that was the case we should just default to true. Maybe something like `It is recommended to enable for this any production databases containing critical data.` instead.